nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/batch/Batch.java

```diff
@@ -55,13 +55,50 @@
 
     protected Map<String, Serializable> fileEntries;
 
+    protected String handlerName;
+
+    protected TransientStore transientStore;
+
+    protected Map<String, Object> properties;
+
     public Batch(String key) {
         this(key, new HashMap<>());
     }
 
     public Batch(String key, Map<String, Serializable> fileEntries) {
+        this(key, fileEntries, null);
+    }
+
+    /**
+     * Constructs a batch.
+     *
+     * @param key the batch key
+     * @param fileEntries the batch file entries
+     * @param handlerName the batch hrovider name
+     * @since 10.1
+     */
+    public Batch(String key, Map<String, Serializable> fileEntries, String handlerName) {
+        this(key, fileEntries, handlerName, null);
+    }
+
+    /**
+     * Constructs a batch.
+     *
+     * @param key the batch key
+     * @param fileEntries the batch file entries
+     * @param handlerName the batch hrovider name
+     * @param transientStore the transient store associated with this Batch
+     * @since 10.1
+     */
+    public Batch(String key, Map<String, Serializable> fileEntries, String handlerName, TransientStore transientStore) {
         this.key = key;
         this.fileEntries = fileEntries;
+        this.handlerName = handlerName;
+        if (transientStore == null) {
+            transientStore = Framework.getService(BatchManager.class).getTransientStore();
+        }
```

Comment:
Services must generally not be cached. Do a lookup at method level when you need it.
This leads to hotreload issues and objects de-synchronization.

Reply:
Indeed services must not be cached.
In this case, the transientStore may be provided by the constructor and the Batch object a short lifecycle, and will never "survive" a hotreload.
IMHO, it's a good trade off to keep this implementation for a better readability.

Reply:
Yes it's all right because the `Batch` class does not live longer than the request.